components/sensor/ld2410.rst

```diff
@@ -114,68 +275,49 @@ and binary sensors.
       - 30
       - 15
 
-Sensor
+Button
 ------
 
-The ``ld2410`` sensor allows you to use your :doc:`ld2410` to perform different
-measurements.
+The ``ld2410`` button allows you to perfrom actions on your :doc:`ld2410`.
 
 .. code-block:: yaml
 
-    sensor:
+    button:
       - platform: ld2410
-        moving_distance:
-          name : Moving Distance
-        still_distance:
-          name: Still Distance
-        moving_energy:
-          name: Move Energy
-        still_energy:
-          name: Still Energy
-        detection_distance:
-          name: Detection Distance
+        factory_reset:
+          name: "factory reset"
+        restart:
+          name: "restart"
+        query_params:
+          name: query params
 
 Configuration variables:
 ************************
 
-- **moving_distance** (*Optional*): Distance of detected moving target.
-  All options from :ref:`Sensor <config-sensor>`.
-- **still_distance** (*Optional*): Distance of detected still target.
-  All options from :ref:`Sensor <config-sensor>`.
-- **moving_energy** (*Optional*): Energy for moving target.
-  All options from :ref:`Sensor <config-sensor>`.
-- **still_energy** (*Optional*): Energy for still target.
-  All options from :ref:`Sensor <config-sensor>`.
-- **detection_distance** (*Optional*): Distance in cm of target.
-  All options from :ref:`Sensor <config-sensor>`.
+- **factory_reset** (*Optional*): Perfrom factory reset.
+  All options from :ref:`Button <config-button>`.
+- **restart** (*Optional*): Restart the device.
+  All options from :ref:`Button <config-button>`.
+- **query_params** (*Optional*): Refresh the :ref:`ld2410-number` values of the device.
+  All options from :ref:`Button <config-button>`.
 
-Binary Sensor
--------------
+Text Sensor
+-----------
 
-The ``ld2410`` binary sensor allows you to use your :doc:`ld2410` to perform different
-measurements.
+The ``ld2410`` text sensor allows you get infomration about your :doc:`ld2410`.
 
 .. code-block:: yaml
 
-    binary_sensor:
+    text_sensor:
       - platform: ld2410
-        has_target:
-          name: Presence
-        has_moving_target:
-          name: Moving Target
-        has_still_target:
-          name: Still Target
+        version:
+          name: "firmware version" 
 
 Configuration variables:
 ************************
 
-- **has_target** (*Optional*): If true target detect either still or in movement.
-  All options from :ref:`Binary Sensor <config-binary_sensor>`.
-- **has_moving_target** (*Optional*): If true a moving target is detected.
-  All options from :ref:`Binary Sensor <config-binary_sensor>`.
-- **has_still_target** (*Optional*): If true a still target is detected.
-  All options from :ref:`Binary Sensor <config-binary_sensor>`.
-
+- **version** (*Optional*): the firmware version.
+  All options from :ref:`Text Sensor <config-text_sensor>`.
 
```

Comment:
Maybe we could add a section explaining setup process, as stated in PR : 

Calibration Process
---------------------

To support the calibration process you have to :

1. enable engineering mode
2. monitor the gates values
3. change the gates thresholds and repeat step 2 if necessary (without reflashing esp)
4. disable engineering mode

Reply:
done